sql/src/main/java/io/crate/execution/engine/collect/ShardCollectorProvider.java

```diff
@@ -112,55 +112,33 @@
         return row;
     }
 
-    /**
-     * Create a CrateCollector.Builder to collect rows from a shard.
-     * <p>
-     * This also creates all shard-level projectors.
-     * The BatchConsumer that is used for {@link CrateCollector.Builder#build(RowConsumer)}
-     * should be the first node-level projector.
-     */
-    public CrateCollector.Builder getCollectorBuilder(RoutedCollectPhase collectPhase,
-                                                      boolean requiresScroll,
-                                                      CollectTask collectTask) throws Exception {
-        assert collectPhase.orderBy() ==
-               null : "getDocCollector shouldn't be called if there is an orderBy on the collectPhase";
-        RoutedCollectPhase normalizedCollectNode = collectPhase.normalize(shardNormalizer, null);
-
-        final CrateCollector.Builder builder;
-        if (!QueryClause.canMatch(normalizedCollectNode.where())) {
-            builder = RowsCollector.emptyBuilder();
-        } else {
-            assert normalizedCollectNode.maxRowGranularity() == RowGranularity.DOC : "granularity must be DOC";
-            builder = getBuilder(normalizedCollectNode, requiresScroll, collectTask);
-        }
+    public BatchIterator<Row> getIterator(RoutedCollectPhase collectPhase,
+                                          boolean requiresScroll,
+                                          CollectTask collectTask) throws Exception {
+        assert collectPhase.orderBy() == null
+            : "getDocCollector shouldn't be called if there is an orderBy on the collectPhase";
```

Comment:
Is this still relevant? (if so it should say getIterator)

Reply:
hmm ... given we have a "getUnorderedIterator" one might assume an orderby here is totally fine no?

Reply:
It's still relevant. Before we had `getCollectorBuilder` and `getBuilder`, both with the same signature. Where one is public and one is internal/for the subclass to override. Not sure how to improve the naming. There is also a public `getOrderedCollector` which is why I went with `getUnorderedIterator` for the protected method.